### DISCUSSION TITLE
Docker-based build script

### DISCUSSION THREAD

**Discussion by git-developer** [v1993/linuxmotehook2#11] [Show and tell]
I'm interested in an improvement of motion control in the retro-gaming distribution [batocera](https://github.com/batocera-linux/batocera.linux/). The Cemuhook protocol seems to be the best choice currently, because some emulators lack SDL motion support (e.g. [Cemu](https://github.com/cemu-project/Cemu/issues/443) - thanks for your active support). I've thrown together a working proof-of-concept, available as [batocera-extra](https://github.com/git-developer/batocera-extra/); there's [a lot of work](https://github.com/git-developer/batocera-extra/issues/1) left, but linuxmotehook2 is a helpful part for the integration of Wimotes ([documentation](https://github.com/git-developer/batocera-extra/blob/main/linuxmotehook2/README.adoc)).

batocera is based on buildroot and does not contain a build toolchain or package manager. Thus, linuxmotehook2 binaries are built separately and bundled with batocera-extra. To simplify this procedure, I wrote a little [script](https://github.com/git-developer/batocera-extra/blob/main/linuxmotehook2/build) which may be run on any Debian-compatible architecture, having Docker as its only dependency. Once you re-enable CI, it could possibly save you some work.

Thanks for linuxmotehook2 and your efforts in making motion control for Wiimotes work on Linux!

P.S.: this issue is meant as notification only, feel free to close it anytime.

**Comment by v1993** [comment]
Nice to know this, thanks! I don't think that script will end up being helpful to me when implementing CI, but it's neat to have it nonetheless.

Also, I got around to release version 0.1.5 - it's a pretty minor bugfix release, but you might want to switch to it.

FYI - Cemu itself does support SDL motion, it's that SDL2 on Linux omits HIDAPI backend by default due to its conflicts with kernel-level drivers without supporting motion from kernel-level drivers ([yuzu overrides this default](https://github.com/yuzu-emu/yuzu/pull/6293)). Worse yet, wiimote's driver does its own thing when it comes to motion due to being developed before proper motion input protocol was.

**Comment by v1993** [comment]
Just for future reference - this kind of conversation is better suited for github discussions than issues.

**Comment by git-developer** [comment]
Thanks a lot for your feedback! Didn't know about GitHub discussions, just the right thing for my intent. You apparently have a lot of experience in this motion control area - thanks for sharing!
